gum/guminterceptor.c

```diff
@@ -905,9 +917,18 @@ _gum_function_context_on_leave (FunctionContext * function_ctx,
     invocation_ctx->backend->data = &state;
 
 #ifdef HAVE_QNX
-    gum_exec_callback_func_with_side_stack (entry->listener_instance,
-        invocation_ctx, entry->listener_interface->on_leave,
-        interceptor_ctx->thread_side_stack + GUM_THREAD_SIDE_STACK_SIZE - 4);
+    gpointer stack_address = &stack_address;
+    if (interceptor_ctx->thread_side_stack < stack_address &&
+        stack_address < interceptor_ctx->thread_side_stack + GUM_THREAD_SIDE_STACK_SIZE)
+    {
+      /* we're already on the side stack, no need to switch. */
+      entry->listener_interface->on_leave (entry->listener_instance,
+          invocation_ctx);
+    }
+    else
+      gum_exec_callback_func_with_side_stack (entry->listener_instance,
```

Comment:
So far the rule of thumb has been that if any of the cases contains a block, all of them do. Also, if a statement is split over multiple lines we always use a block.